### PR TITLE
Replace WebClient usage with HttpClient

### PR DESCRIPTION
**Pull request by ericstj**
https://github.com/dotnet/machinelearning/pull/6385 and https://github.com/dotnet/machinelearning/pull/6367 clashed after merge.  Fix the build break.

### CODE REVIEW

File: test/Microsoft.ML.AutoML.Tests/DatasetUtil.cs [hunk 1]
```diff
@@ -203,7 +205,16 @@ private static void Download(string url, string destDir, string destFileName)
             if (File.Exists(relativeFilePath))
                 return;
 
-            new WebClient().DownloadFile(url, relativeFilePath);
+            using (var client = new HttpClient())
+            {
+                var response = await client.GetAsync(url);
```
Comment (tarekgh) [inline test/Microsoft.ML.AutoML.Tests/DatasetUtil.cs:210]: >response [](http://example.com/codeflow?start=20&length=8)

do we need to dispose this?

**Review by tarekgh — APPROVED** [review]

**Comment by ericstj** [comment]
Unix failure is System.Drawing:
```
   Microsoft.ML.Tests.ImageTests.TestBackAndForthConversionWithAlphaInterleave [FAIL]
[31;1m[m[37m      System.ArgumentException : Parameter is not valid.
[m[30;1m      Stack Trace:
[m[37m           at System.Drawing.SafeNativeMethods.Gdip.CheckStatus(Int32 status)
[m[37m           at System.Drawing.Bitmap.UnlockBits(BitmapData bitmapdata)
[m[37mStarting test: Microsoft.ML.Tests.ImageTests.TestSaveImages
        /__w/1/s/src/Microsoft.ML.ImageAnalytics/VectorToImageTransform.cs(421,0): at Microsoft.ML.Transforms.Image.VectorToImageConvertingTransformer.Mapper.<>c__DisplayClass5_0`1.<GetterFromType>b__0(Bitmap& dst)
[m[37m        /__w/1/s/test/Microsoft.ML.Tests/ImagesTests.cs(345,0): at Microsoft.ML.Tests.ImageTests.TestBackAndForthConversionWithAlphaInterleave()
```

File: test/Microsoft.ML.AutoML.Tests/DatasetUtil.cs (hunk 1 above)
Comment (ericstj) [reply test/Microsoft.ML.AutoML.Tests/DatasetUtil.cs:210]: Maybe?  I copied this from elsewhere in the repo which doesn’t dispose.  Perhaps disposing the wrapping client takes care of any open resources.  I will look into it and follow up with all the instances of the problem in the repo if necessary. 
https://github.com/dotnet/machinelearning/blob/e8073ad4eb59e0367a9c64f6f607f301282b2591/test/Microsoft.ML.Tokenizers.Tests/Utils.cs#L16
https://github.com/dotnet/machinelearning/search?q=httpclient+getasync&type=

**Comment by ericstj** [comment]
Definitely broke Linux, it’s hanging.  Will see what I did wrong. Don’t merge please.

**Comment by tarekgh** [comment]
Could be `.Wait()` call? using `await` worth to try.

**Comment by ericstj** [comment]
Can’t await unless I change the whole stack. ConfigureAwait is the solution.

**Pull request merged**